src/components/shared/Visualizer.vue

```diff
@@ -131,11 +134,18 @@ export default {
       return undefined;
     },
     tree() {
-      return d3.tree().size([1000, this.settings.width - 300]).separation(() => (this.settings.keyCellWidth * 2));
-    },
-    getWith() {
-      return this.$refs && this.$refs.cont ? this.$refs.cont.clientWidth : 800;
+      return d3.tree().size([this.svgWidth, this.settings.width - 300]).separation(() => (this.settings.keyCellWidth * 2));
     },
+
+  },
+
+
+  mounted() {
+    this.addSizingListeners();
+  },
+
+  beforeDestroy() {
+    this.removeListeners();
```

Comment:
Excellent. But maybe call directly `window.onresize = null;` here and comment `// Remove listeners`

Reply:
I think it makes for a nice encapsulation. If we ever have other listeners (ad I think we'll do when trying to apply scroll directives to the visualizer), this method will be open for extension. What do you think?

Reply:
If we will call this.removeListeners() just before destroying the component, I do not see the benefit of doing an extra encapsulation. If it gonna be useful during the lifecycle of the component in other occasion, I will be agree